text_generation/causal_lm/cpp/continuous_batching/library/src/scheduler.hpp

```diff
@@ -219,15 +260,20 @@ class Scheduler {
                 size_t num_scheduled_tokens_per_seq = std::min(available_tokens_per_seq_in_megabatch, num_available_tokens_per_seq);
                 sequence_group->schedule_tokens(num_scheduled_tokens_per_seq);
 
-                // TODO: below functions can_append_slot / append_slot can allocate just a single slot, while we require multiple ones in generic case
-                // (generic case is speculative deconding, where we can check multiple tokens at once)
-                // So, let's state this as current limitation of scheduler logic
-                OPENVINO_ASSERT(num_scheduled_tokens_per_seq == 1);
+                // check that all scheduled tokens can be allocated and apply preemtion if needed
+                while (!m_block_manager.can_append_slots(sequence_group)) {
+                    if (!_apply_preemption(sequence_group_id, sequence_groups))
+                        break;
+                }
```

Comment:
can we hide this cycle to `_apply_preemption`?
in this case scheduler logic will just call `_apply_preemption(sequence_group_id, sequence_groups)` without return values

Reply:
`_apply_preemption` already has cycle that runs until `can_append_slots()` returns true. Looks like second cycle is redundant, so I removed it.